ProcessOptimizer/model_systems/__init__.py

```diff
@@ -0,0 +1,31 @@
+from .branin_hoo import branin_hoo
+from .model_system import ModelSystem
+from .noise_models import (
+    DataDependentNoise,
+    ZeroNoise,
+    ConstantNoise,
+    ProportionalNoise,
+    SumNoise,
+    parse_noise_model,
+    noise_model_factory,
+)
+from .benchmarks import hart3_model_system as hart3
```

Comment:
This isn't very nice - I am importing the model system, and renaming it to have the same name as the benchmark function has in `benchmarks.py`. This could cause confusion, as
````
from model_systems import hart3
```
and
```
from model_systems.benchmarks import hart3
```
returns very different objects named `hart3`, and it is not super transparent why.

I want the user to automatically get the model system, and not even notice that there is a benchmark function, but also keep the benchmark function to be imported exactly as is, for backward compatibility. Please comment if my approach is acceptable.

We could also simply
```
from .benchmarks import hart3_model_system
```
, but I would like the "thing" the user gets to be a model system that is named after the benchmark function.

Reply:
Hmm, I'm not convinced that we have a plethora of users that make extensive use of the benchmark _functions_ themselves yet, mainly because most of them are quite recent additions (peaks + hartmann functions). Couldn't we just rename the functions inside `benchmarks.py` to `hart3_function`, `peaks_function` etc. since this would make the distinction explicit and the code that builds the model systems more readable? Then we can keep the short names like `hart3` reserved for the thing that we intend for the users to interact with, namely the model systems.

Reply:
So the consequence would be to expand the names of the soon-to-be entirely internal function definitions inside benchmarks.py and shorten the names of the actual model systems down.

Reply:
If we are to change that, we could also consider splitting them into separate files, like the one named `gold_map.py`. It is easier to find them that way. It doesn't change anything for the user, since they are still imported into `__init__.py`, so it would still be `from ProcessOptimizer.model_systems import XXXX`.

We get quite a lot of files in that folder, but I don't see that as a big issue.

Reply:
I don't mind that implementation either, so feel free to go for that :)

Reply:
I will, then, but I won't have time before my holiday. I will get on it in August.

Reply:
@dk-teknologisk-mon : Done, would you give it a new review?